From 4ca727fd409bad029ce83debe43699a070d25bba Mon Sep 17 00:00:00 2001
From: Arun Kumar <arunkumar.ev@adlinktech.com>
Date: Mon, 20 Apr 2020 19:22:39 +0530
Subject: [PATCH 08/11] IMX8QM: Add dts file for ov5640

Add dts file for ov5640 camera sensor based on freescale
ov5640 dts. Disable i2c0_mipi_csi1.

Signed-off-by: Arun Kumar <arunkumar.ev@adlinktech.com>
---
 arch/arm64/boot/dts/adlink/Makefile           |   3 +-
 .../dts/adlink/adlink-lec-imx8qm-ov5640.dts   | 108 ++++++++++++++++++
 2 files changed, 110 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/adlink/adlink-lec-imx8qm-ov5640.dts

diff --git a/arch/arm64/boot/dts/adlink/Makefile b/arch/arm64/boot/dts/adlink/Makefile
index a418d0fd41e2..4aa9fc65a5bd 100644
--- a/arch/arm64/boot/dts/adlink/Makefile
+++ b/arch/arm64/boot/dts/adlink/Makefile
@@ -1,5 +1,6 @@
 dtb-$(CONFIG_ARCH_LEC_IMX8QM) += adlink-lec-imx8qm.dtb \
-				adlink-lec-imx8qm-hdmi.dtb
+				adlink-lec-imx8qm-hdmi.dtb \
+				adlink-lec-imx8qm-ov5640.dtb
 
 always         := $(dtb-y)
 subdir-y       := $(dts-dirs)
diff --git a/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm-ov5640.dts b/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm-ov5640.dts
new file mode 100644
index 000000000000..9ff7ba588fd2
--- /dev/null
+++ b/arch/arm64/boot/dts/adlink/adlink-lec-imx8qm-ov5640.dts
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2018 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "adlink-lec-imx8qm.dts"
+
+&iomuxc {
+	imx8qm-mek {
+		pinctrl_mipi_csi0: mipicsi0grp{
+			fsl,pins = <
+				/*SC_P_MIPI_CSI0_MCLK_OUT_MIPI_CSI0_ACM_MCLK_OUT	0xC0000041*/
+				SC_P_MIPI_CSI0_GPIO0_00_LSIO_GPIO1_IO27			0x00000021
+				SC_P_MIPI_CSI0_GPIO0_01_LSIO_GPIO1_IO28			0x00000021
+			>;
+		};
+	};
+};
+
+&isi_0 {
+	status = "okay";
+};
+
+&isi_4 {
+	status = "disabled";
+};
+
+&isi_1 {
+	status = "disabled";
+};
+
+&isi_2 {
+	status = "disabled";
+};
+
+&isi_3 {
+	status = "disabled";
+};
+
+&isi_5 {
+	status = "disabled";
+};
+
+&isi_6 {
+	status = "disabled";
+};
+
+&isi_7 {
+	status = "disabled";
+};
+
+&i2c0_mipi_csi0 {
+	clock-frequency = <100000>;
+	status = "okay";
+
+	ov5640_mipi: ov5640_mipi@3c {
+		compatible = "ovti,ov5640_mipi_v3";
+		reg = <0x3c>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_mipi_csi0>;
+		clocks = <&clk IMX8QM_CLK_DUMMY>;
+		clock-names = "csi_mclk";
+		csi_id = <0>;
+		pwn-gpios = <&gpio1 28 GPIO_ACTIVE_LOW>;
+		rst-gpios = <&gpio1 27 GPIO_ACTIVE_HIGH>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		mipi_csi;
+		status = "okay";
+		port {
+			ov5640_mipi_ep: endpoint {
+				remote-endpoint = <&mipi_csi0_ep>;
+			};
+		};
+	};
+
+	/delete-node/max9286_mipi@6A;
+};
+
+&i2c0_mipi_csi1 {
+	status = "disabled";
+};
+
+&mipi_csi_0 {
+	/delete-property/virtual-channel;
+	status = "okay";
+
+	port@0 {
+		reg = <0>;
+		mipi_csi0_ep: endpoint {
+			remote-endpoint = <&ov5640_mipi_ep>;
+			data-lanes = <1 2>;
+		};
+	};
+};
+
+&mipi_csi_1 {
+	status = "disabled";
+};
-- 
2.17.1

